### PR TITLE
Fix CVSS v2 parsing and improve performances

### PR DESCRIPTION
**Pull request by pandatix**
Fix issue #50481 and improve performances by using standard string index-based matching and cut.

## Test plan

- Added unit tests for `parseCVSS()`

### CODE REVIEW

**Comment by pandatix** [comment]
@sourcegraph/contribution-reviewers

**Comment by pandatix** [comment]
@sourcegraph/contribution-reviewers

Any news ? Should be interesting to fix before CVSS v4.0 release (october) and unecessary rebases :)

**Comment by jdorfman** [comment]
Thanks @pandatix for the ping

cc: @andreeleuterio

**Review by willdollman — APPROVED** [review]
lgtm, thanks for this @pandatix !

File: enterprise/internal/codeintel/sentinel/internal/background/downloader/source_osv.go [hunk 1]
```diff
@@ -8,7 +8,6 @@ import (
 	"github.com/sourcegraph/log"
 
 	"github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/sentinel/shared"
```
Comment (willdollman) [inline enterprise/internal/codeintel/sentinel/internal/background/downloader/source_osv.go:10]:
```suggestion
	"github.com/sourcegraph/sourcegraph/internal/codeintel/sentinel/shared"
```

**Pull request merged**